community/cypher/cypher/src/main/scala/org/neo4j/cypher/internal/spi/v3_0/TransactionBoundQueryContext.scala

```diff
@@ -64,8 +64,9 @@ final class TransactionBoundQueryContext(val transactionalContext: Transactional
 
   val nodeOps = new NodeOperations
   val relationshipOps = new RelationshipOperations
+  lazy val nodeManager = transactionalContext.graph.getDependencyResolver.resolveDependency(classOf[NodeManager])
 
-  override def entityAccessor = transactionalContext.graph.getDependencyResolver.resolveDependency(classOf[NodeManager])
+  override def entityAccessor = nodeManager
```

Comment:
why not `override lazy val entityAccessor = transactionalContext.graph.getDependencyResolver.resolveDependency(classOf[NodeManager])`?